VideoApi/VideoApi.Events/Handlers/DisconnectedEventHandler.cs

```diff
@@ -40,7 +42,7 @@ private async Task PublishParticipantDisconnectMessage()
 
         private async Task AddDisconnectedTask()
         {
-            var taskType = SourceParticipant.IsJudge() ? TaskType.Judge : TaskType.Participant;
+            var taskType = SourceParticipant is Participant && ((Participant)SourceParticipant).IsJudge() ? TaskType.Judge : TaskType.Participant;
```

Comment:
Seeing this pattern of checking if something is participant and then casting.

Should we have these IsJudge() and similar methods as part of ParticipantBase or maybe a helper class to do these checks.